### PR TITLE
Refactor mission transfer

### PR DESCRIPTION
**Pull request by julianoes**
This moves mission/geofence/rally point transfer into core which means the work that has to be done in the plugins can be simplified

### CODE REVIEW

**Comment by julianoes** [comment]
Try again tomorrow, now libressl download is down:
```
 -- Using src='https://ftp.openbsd.org/pub/OpenBSD/LibreSSL/libressl-3.0.2.tar.gz'
CMake Error at libressl-stamp/download-libressl.cmake:159 (message):
```

**Comment by julianoes** [comment]
@JonasVautherin ready for review :smile:

File: src/plugins/geofence/geofence_impl.cpp [hunk 1]
```diff
@@ -183,64 +55,70 @@ void GeofenceImpl::assemble_mavlink_messages(
...
+Geofence::Result GeofenceImpl::convert_result(MAVLinkMissionTransfer::Result result)
 {
-    if (callback == nullptr) {
-        LogDebug() << "Callback is not set";
-        return;
+    switch (result) {
+        case MAVLinkMissionTransfer::Result::Success:
+            return Geofence::Result::SUCCESS;
+        case MAVLinkMissionTransfer::Result::ConnectionError:
+            return Geofence::Result::ERROR; // FIXME
```
Comment (JonasVautherin) [inline src/plugins/geofence/geofence_impl.cpp]: How should those be fixed? By adding Geofence results?

Comment (julianoes) [reply src/plugins/geofence/geofence_impl.cpp]: I planned to consolidate the different results in a future PR. I think the result type will be duplicated / or similar between the plugins but we will add some more error types that make sense. My goal was basically to not have any change in API for this PR but only swap out internals.

**Comment by julianoes** [comment]
Thanks for merging. I didn't have the courage :smile:.

**Comment by julianoes** [comment]
FYI @hamishwillee we're on the way with the plan we decided on a while ago :tada:

**Comment by itrigitech** [comment]
Oh perfect. Thanks for that work! In March then I will be able to work in the needed plugins

**Comment by hamishwillee** [comment]
Well done!

**Pull request merged**